File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/linqexamples52.vb
' <snippetlinqexamples52>


 Using svcContext As New ServiceContext(_serviceProxy)
  Dim query_twowhere = From a In svcContext.AccountSet _
                       Join c In svcContext.ContactSet _
                       On a.PrimaryContactId.Id Equals c.ContactId _
                       Where c.LastName.Equals("Smith") _
                       AndAlso c.CreditOnHold IsNot Nothing _
                       Where a.Name.Equals("Contoso Ltd") _
                       Order By a.Name _
                       Select a
  For Each c In query_twowhere
   Console.WriteLine(c.AccountId.ToString() &amp; " " &amp; c.Name)
  Next c
 End Using

' </snippetlinqexamples52>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/linqexamples52.cs
// <snippetlinqexamples52>


 using (ServiceContext svcContext = new ServiceContext(_serviceProxy))
 {
  var query_twowhere = from a in svcContext.AccountSet
                       join c in svcContext.ContactSet 
                       on a.PrimaryContactId.Id equals c.ContactId
                       where c.LastName == "Smith" && c.CreditOnHold != null
                       where a.Name == "Contoso Ltd"
                       orderby a.Name
                       select a;
  foreach (var c in query_twowhere)
  {
   System.Console.WriteLine(c.AccountId + " " + c.Name);
  }
 }

// </snippetlinqexamples52>